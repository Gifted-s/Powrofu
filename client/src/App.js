import React from 'react';
import SignUp from './Components/Signup/Signup';
import { BrowserRouter, Route, Switch } from 'react-router-dom'
import DashBoard from './Components/Dashboard/Dashboard';
import AuthContextProvider from './contexts/AuthContext';

function App() {

  return (
    <BrowserRouter>

      <Switch>
        <AuthContextProvider>
          <Route path="/" component={SignUp} exact />
          <Route path="/dashboard" component={DashBoard} />
        </AuthContextProvider>
      </Switch>

    </BrowserRouter>

  )
}
export default App;
