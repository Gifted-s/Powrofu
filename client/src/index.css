.form-container{
  padding-top:20px;
  padding:15px;
  
 }
 .btn{
   width:50%;
 }
 .dashboard-container{
   padding: 30px
 }
 .search_item:hover{
   background-color: rgba(0,0,0,0.1);
 }

 @media screen and (max-width:768px){
   .dashboard-container{
     padding:0
   }
 }
 h3{
   text-align:center
 }
 .location_list{
   height:60px;
 }
 .location_list:hover{
  background-color: rgba(9, 102, 131, 0.04);
 }