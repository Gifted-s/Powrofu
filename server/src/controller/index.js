const signUp = require('../use-cases/index')
const makePostUser = require('./post-user')
const postUser  = makePostUser({signUp})

const controllers = Object.freeze({
    postUser
})

module.exports= controllers