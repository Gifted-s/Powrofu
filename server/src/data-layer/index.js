const  makeUsersDb = require( './userDB')
const  mongodb  = require('mongodb') 
const  dotenv = require('dotenv')
dotenv.config()
const MongoClient = mongodb.MongoClient
const dbName = process.env.DB_NAME
const client = new MongoClient(process.env.MONGODB_CONN, { useNewUrlParser: true, useUnifiedTopology:true }, ()=> console.log('db connected'))
 async function makeDb () {
  if (!client.isConnected()) {
    await client.connect()
  }
  return client.db(dbName)
}
const User = makeUsersDb({ makeDb })
module.exports =  User
