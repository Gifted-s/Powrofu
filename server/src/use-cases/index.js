const User = require("../data-layer");
const buildSignUp = require('./sign-up')
const  signUp = buildSignUp({User})
module.exports = signUp