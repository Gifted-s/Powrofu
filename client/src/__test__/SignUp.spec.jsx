import React from 'react';
import { Route, BrowserRouter, Switch } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { render,fireEvent } from '@testing-library/react';
import { default as fetcher } from 'node-fetch';
import SignUp from '../Components/Signup/Signup';
import DashBoard from '../Components/Dashboard/Dashboard';
import AuthContextProvider from '../contexts/AuthContext';
import '../../src/setupTests'
jest.mock('node-fetch');
const { Response } = jest.requireActual('node-fetch');
const componentTest = () => {
  return (
    <BrowserRouter>

      <Switch>
        <AuthContextProvider>
          <Route path="/" component={SignUp} exact />
          <Route path="/dashboard" component={DashBoard} />
        </AuthContextProvider>
      </Switch>

    </BrowserRouter>
  );
};

test('Must render properly', () => {
  const tree = renderer
    .create(componentTest())
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('it cannot signup without values',async () => {
  const formInput= {
    email: '',
    password:'',
    location:'',
    firstname:'',
    gender:''
  
  };
  const { getByLabelText, getByText} = render(componentTest());
     fireEvent.change(getByLabelText(/First name/i), {target:{value:formInput.firstname}})
     fireEvent.change(getByLabelText(/Gender/i), {target:{value:formInput.gender}})
     fireEvent.change(getByLabelText(/E-mail/i), {target:{value:formInput.email}})
     fireEvent.change(getByLabelText(/Enter your location/i), {target:{value:formInput.location}})
     fireEvent.change(getByLabelText('Password'), {target:{value:formInput.password}})
     fireEvent.click(getByText(/Signup/i))
     expect(()=>getByText(/Please input your firstname/i)).toBeTruthy()
     expect(()=>getByText(/Gender is required/i)).toBeTruthy()
     expect(()=>getByText(/Please input your E-mail!/i)).toBeTruthy()
     expect(()=>getByText(/Please type you location here/i)).toBeTruthy()
     expect(()=>getByText(/Please input your password!/i)).toBeTruthy()
});



test('it cannot signup without email',async () => {
    const formInput= {
      email: '',
      password:'Ayodeji00;',
      location:'Ado Ekiti Lat 34^ Log40^',
      firstname:'Adewumi',
      gender:'Male'
    };
    const { getByLabelText, getByText} = render(componentTest());
       fireEvent.change(getByLabelText(/First name/i), {target:{value:formInput.firstname}})
       fireEvent.change(getByLabelText(/Gender/i), {target:{value:formInput.gender}})
       fireEvent.change(getByLabelText(/E-mail/i), {target:{value:formInput.email}})
       fireEvent.change(getByLabelText(/Enter your location/i), {target:{value:formInput.location}})
       fireEvent.change(getByLabelText('Password'), {target:{value:formInput.password}})
       fireEvent.click(getByText(/Signup/i))
       expect(()=>getByText(/Please input your E-mail!/i)).toBeTruthy()
      
  });
  
test('it cannot signup without password',async () => {
    const formInput= {
      email: 'test@gmail.com',
      password:'',
      location:'Ado Ekiti Lat 34^ Log40^',
      firstname:'Adewumi',
      gender:'Male'
    };
    const { getByLabelText, getByText} = render(componentTest());
       fireEvent.change(getByLabelText(/First name/i), {target:{value:formInput.firstname}})
       fireEvent.change(getByLabelText(/Gender/i), {target:{value:formInput.gender}})
       fireEvent.change(getByLabelText(/E-mail/i), {target:{value:formInput.email}})
       fireEvent.change(getByLabelText(/Enter your location/i), {target:{value:formInput.location}})
       fireEvent.change(getByLabelText('Password'), {target:{value:formInput.password}})
       fireEvent.click(getByText(/Signup/i))
       expect(()=>getByText(/Please input your password!/i)).toBeTruthy()
  });
  
  test('it cannot signup without location',async () => {
    const formInput= {
      email: 'test@gmail.com',
      password:'Ayodeji00;',
      location:'',
      firstname:'Adewumi',
      gender:'Male'
    };
    const { getByLabelText, getByText} = render(componentTest());
       fireEvent.change(getByLabelText(/First name/i), {target:{value:formInput.firstname}})
       fireEvent.change(getByLabelText(/Gender/i), {target:{value:formInput.gender}})
       fireEvent.change(getByLabelText(/E-mail/i), {target:{value:formInput.email}})
       fireEvent.change(getByLabelText(/Enter your location/i), {target:{value:formInput.location}})
       fireEvent.change(getByLabelText('Password'), {target:{value:formInput.password}})
       fireEvent.click(getByText(/Signup/i))
       expect(()=>getByText(/Please type you location here/i)).toBeTruthy()
  });
  test('it cannot signup without firstname',async () => {
    const formInput= {
      email: 'test@gmail.com',
      password:'Ayodeji00;',
      location:'Ado Ekiti Lat 34^ Log40^',
      firstname:'',
      gender:'Male'
    };
    const { getByLabelText, getByText} = render(componentTest());
       fireEvent.change(getByLabelText(/First name/i), {target:{value:formInput.firstname}})
       fireEvent.change(getByLabelText(/Gender/i), {target:{value:formInput.gender}})
       fireEvent.change(getByLabelText(/E-mail/i), {target:{value:formInput.email}})
       fireEvent.change(getByLabelText(/Enter your location/i), {target:{value:formInput.location}})
       fireEvent.change(getByLabelText('Password'), {target:{value:formInput.password}})
       fireEvent.click(getByText(/Signup/i))
       expect(()=>getByText(/Please input your firstname/i)).toBeTruthy()
  });

  test('it cannot signup without gender',async () => {
    const formInput= {
      email: 'test@gmail.com',
      password:'Ayodeji00;',
      location:'Ado Ekiti Lat 34^ Log40^',
      firstname:'Adewumi',
      gender:''
    };
    const { getByLabelText, getByText} = render(componentTest());
       fireEvent.change(getByLabelText(/First name/i), {target:{value:formInput.firstname}})
       fireEvent.change(getByLabelText(/Gender/i), {target:{value:formInput.gender}})
       fireEvent.change(getByLabelText(/E-mail/i), {target:{value:formInput.email}})
       fireEvent.change(getByLabelText(/Enter your location/i), {target:{value:formInput.location}})
       fireEvent.change(getByLabelText('Password'), {target:{value:formInput.password}})
       fireEvent.click(getByText(/Signup/i))
       expect(()=>getByText(/Gender is required/i)).toBeTruthy()
  });
  
  
  
test('it can signup with the right credentials',async () => {
  const body = {
    user: {
        email: 'uniquetestemail@gmail.com',
        password:'Ayodeji00;',
        location:'Ado Ekiti Lat 34^ Log40^',
        firstName:'Adewumi',
        gender:'male'
    },
    status: 'success',
  };
  // mock respone
  fetcher.mockReturnValue(Promise.resolve(new Response(JSON.stringify(body))));

  const formInput= {
    email: 'uniquetestemail@gmail.com',
    password:'Ayodeji00;',
    location:'Ado Ekiti Lat 34^ Log40^',
    firstname:'Adewumi',
    gender:'male'
  };
  const { getByLabelText, getByText } = render(componentTest());
      fireEvent.change(getByLabelText(/First name/i), {target:{value:formInput.firstname}})
      fireEvent.change(getByLabelText(/Gender/i), {target:{value:formInput.gender}})
      fireEvent.change(getByLabelText(/E-mail/i), {target:{value:formInput.email}})
      fireEvent.change(getByLabelText(/Enter your location/i), {target:{value:formInput.location}})
      fireEvent.change(getByLabelText('Password'), {target:{value:formInput.password}})
     fireEvent.click(getByText(/Signup/i))
     expect(()=>getByText(/Welcome Adewumi/i)).toBeTruthy()
 
});
