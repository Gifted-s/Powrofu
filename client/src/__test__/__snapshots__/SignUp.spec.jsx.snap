// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Must render properly 1`] = `
<div
  className="form-container"
>
  <h3>
    sign up to have an account
  </h3>
  <form
    className="ant-form ant-form-vertical"
    id="register"
    onSubmit={[Function]}
  >
    
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-label ant-col-xs-8 ant-col-xs-offset-0 ant-col-sm-8 ant-col-sm-offset-8"
        style={Object {}}
      >
        <label
          className="ant-form-item-required"
          htmlFor="register_firstname"
          title="First name"
        >
          First name
        </label>
      </div>
      <div
        className="ant-col ant-form-item-control ant-col-xs-8 ant-col-xs-offset-0 ant-col-sm-8 ant-col-sm-offset-8"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <span
              className="ant-input-affix-wrapper"
              onMouseUp={[Function]}
            >
              <span
                className="ant-input-prefix"
              >
                <span
                  aria-label="user"
                  className="anticon anticon-user site-form-item-icon"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="user"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                    />
                  </svg>
                </span>
              </span>
              <input
                className="ant-input"
                id="register_firstname"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="First name"
                style={null}
                type="text"
                value=""
              />
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-label ant-col-xs-8 ant-col-xs-offset-0 ant-col-sm-8 ant-col-sm-offset-8"
        style={Object {}}
      >
        <label
          className="ant-form-item-required"
          htmlFor="register_gender"
          title="Gender"
        >
          Gender
        </label>
      </div>
      <div
        className="ant-col ant-form-item-control ant-col-xs-8 ant-col-xs-offset-0 ant-col-sm-8 ant-col-sm-offset-8"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <div
              className="ant-select ant-select-single ant-select-allow-clear ant-select-show-arrow"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
            >
              <div
                className="ant-select-selector"
                onClick={[Function]}
                onMouseDown={[Function]}
              >
                <span
                  className="ant-select-selection-search"
                >
                  <input
                    aria-activedescendant="register_gender_list_0"
                    aria-autocomplete="list"
                    aria-controls="register_gender_list"
                    aria-haspopup="listbox"
                    aria-owns="register_gender_list"
                    autoComplete="off"
                    className="ant-select-selection-search-input"
                    id="register_gender"
                    onChange={[Function]}
                    onCompositionEnd={[Function]}
                    onCompositionStart={[Function]}
                    onKeyDown={[Function]}
                    onMouseDown={[Function]}
                    onPaste={[Function]}
                    readOnly={true}
                    role="combobox"
                    style={
                      Object {
                        "opacity": 0,
                      }
                    }
                    type="search"
                    unselectable="on"
                    value=""
                  />
                </span>
                <span
                  className="ant-select-selection-placeholder"
                >
                  Select your gender
                </span>
              </div>
              <span
                aria-hidden={true}
                className="ant-select-arrow"
                onMouseDown={[Function]}
                style={
                  Object {
                    "WebkitUserSelect": "none",
                    "userSelect": "none",
                  }
                }
                unselectable="on"
              >
                <span
                  aria-label="down"
                  className="anticon anticon-down ant-select-suffix"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="down"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                    />
                  </svg>
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-label ant-col-xs-8 ant-col-xs-offset-0 ant-col-sm-8 ant-col-sm-offset-8"
        style={Object {}}
      >
        <label
          className="ant-form-item-required"
          htmlFor="register_location"
          title="Enter your location"
        >
          Enter your location
        </label>
      </div>
      <div
        className="ant-col ant-form-item-control ant-col-xs-8 ant-col-xs-offset-0 ant-col-sm-8 ant-col-sm-offset-8"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <span
              className="ant-input-affix-wrapper ant-input-affix-wrapper-disabled"
              onMouseUp={[Function]}
            >
              <span
                className="ant-input-prefix"
              >
                <span
                  aria-label="environment"
                  className="anticon anticon-environment site-form-item-icon"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="environment"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M854.6 289.1a362.49 362.49 0 00-79.9-115.7 370.83 370.83 0 00-118.2-77.8C610.7 76.6 562.1 67 512 67c-50.1 0-98.7 9.6-144.5 28.5-44.3 18.3-84 44.5-118.2 77.8A363.6 363.6 0 00169.4 289c-19.5 45-29.4 92.8-29.4 142 0 70.6 16.9 140.9 50.1 208.7 26.7 54.5 64 107.6 111 158.1 80.3 86.2 164.5 138.9 188.4 153a43.9 43.9 0 0022.4 6.1c7.8 0 15.5-2 22.4-6.1 23.9-14.1 108.1-66.8 188.4-153 47-50.4 84.3-103.6 111-158.1C867.1 572 884 501.8 884 431.1c0-49.2-9.9-97-29.4-142zM512 880.2c-65.9-41.9-300-207.8-300-449.1 0-77.9 31.1-151.1 87.6-206.3C356.3 169.5 431.7 139 512 139s155.7 30.5 212.4 85.9C780.9 280 812 353.2 812 431.1c0 241.3-234.1 407.2-300 449.1zm0-617.2c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm79.2 255.2A111.6 111.6 0 01512 551c-29.9 0-58-11.7-79.2-32.8A111.6 111.6 0 01400 439c0-29.9 11.7-58 32.8-79.2C454 338.6 482.1 327 512 327c29.9 0 58 11.6 79.2 32.8C612.4 381 624 409.1 624 439c0 29.9-11.6 58-32.8 79.2z"
                    />
                  </svg>
                </span>
              </span>
              <input
                className="ant-input ant-input-disabled"
                disabled={true}
                id="register_location"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="Type your location here"
                style={null}
                type="text"
                value=""
              />
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-col ant-col-sm-8 ant-col-sm-offset-8"
      style={Object {}}
    />
    <div
      className="ant-row ant-form-item input"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-label ant-col-xs-8 ant-col-xs-offset-0 ant-col-sm-8 ant-col-sm-offset-8"
        style={Object {}}
      >
        <label
          className="ant-form-item-required"
          htmlFor="register_email"
          title="E-mail"
        >
          E-mail
        </label>
      </div>
      <div
        className="ant-col ant-form-item-control ant-col-xs-8 ant-col-xs-offset-0 ant-col-sm-8 ant-col-sm-offset-8"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <span
              className="ant-input-affix-wrapper"
              onMouseUp={[Function]}
            >
              <span
                className="ant-input-prefix"
              >
                <span
                  aria-label="mail"
                  className="anticon anticon-mail site-form-item-icon"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="mail"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M928 160H96c-17.7 0-32 14.3-32 32v640c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V192c0-17.7-14.3-32-32-32zm-40 110.8V792H136V270.8l-27.6-21.5 39.3-50.5 42.8 33.3h643.1l42.8-33.3 39.3 50.5-27.7 21.5zM833.6 232L512 482 190.4 232l-42.8-33.3-39.3 50.5 27.6 21.5 341.6 265.6a55.99 55.99 0 0068.7 0L888 270.8l27.6-21.5-39.3-50.5-42.7 33.2z"
                    />
                  </svg>
                </span>
              </span>
              <input
                className="ant-input"
                id="register_email"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="Enter your email"
                style={null}
                type="text"
                value=""
              />
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-label ant-col-xs-8 ant-col-xs-offset-0 ant-col-sm-8 ant-col-sm-offset-8"
        style={Object {}}
      >
        <label
          className="ant-form-item-required"
          htmlFor="register_password"
          title="Password"
        >
          Password
        </label>
      </div>
      <div
        className="ant-col ant-form-item-control ant-col-xs-8 ant-col-xs-offset-0 ant-col-sm-8 ant-col-sm-offset-8"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <span
              className="ant-input-affix-wrapper ant-input-password"
              onMouseUp={[Function]}
            >
              <span
                className="ant-input-prefix"
              >
                <span
                  aria-label="lock"
                  className="anticon anticon-lock site-form-item-icon"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="lock"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32zM332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332V240zm460 600H232V536h560v304zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 10-56 0z"
                    />
                  </svg>
                </span>
              </span>
              <input
                action="click"
                className="ant-input"
                id="register_password"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="Enter your password"
                style={null}
                type="password"
                value=""
              />
              <span
                className="ant-input-suffix"
              >
                <span
                  aria-label="eye-invisible"
                  className="anticon anticon-eye-invisible ant-input-password-icon"
                  onClick={[Function]}
                  onMouseDown={[Function]}
                  onMouseUp={[Function]}
                  role="img"
                  tabIndex={-1}
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="eye-invisible"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                    />
                    <path
                      d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                    />
                  </svg>
                </span>
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-label ant-col-xs-8 ant-col-xs-offset-0 ant-col-sm-8 ant-col-sm-offset-8"
        style={Object {}}
      >
        <label
          className="ant-form-item-required"
          htmlFor="register_confirm"
          title="Confirm Password"
        >
          Confirm Password
        </label>
      </div>
      <div
        className="ant-col ant-form-item-control ant-col-xs-8 ant-col-xs-offset-0 ant-col-sm-8 ant-col-sm-offset-8"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <span
              className="ant-input-affix-wrapper ant-input-password"
              onMouseUp={[Function]}
            >
              <span
                className="ant-input-prefix"
              >
                <span
                  aria-label="lock"
                  className="anticon anticon-lock site-form-item-icon"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="lock"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32zM332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332V240zm460 600H232V536h560v304zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 10-56 0z"
                    />
                  </svg>
                </span>
              </span>
              <input
                action="click"
                className="ant-input"
                id="register_confirm"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="Retype your password"
                style={null}
                type="password"
                value=""
              />
              <span
                className="ant-input-suffix"
              >
                <span
                  aria-label="eye-invisible"
                  className="anticon anticon-eye-invisible ant-input-password-icon"
                  onClick={[Function]}
                  onMouseDown={[Function]}
                  onMouseUp={[Function]}
                  role="img"
                  tabIndex={-1}
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="eye-invisible"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                    />
                    <path
                      d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                    />
                  </svg>
                </span>
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-control ant-col-xs-8 ant-col-xs-offset-8 ant-col-sm-16 ant-col-sm-offset-8"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <button
              className="ant-btn ant-btn-primary btn"
              onClick={[Function]}
              type="submit"
            >
              <span>
                Signup
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
